src/main/scala/com/databricks/labs/overwatch/pipeline/DbsqlTransforms.scala

```diff
@@ -0,0 +1,130 @@
+package com.databricks.labs.overwatch.pipeline
+
+import com.databricks.labs.overwatch.pipeline.PipelineFunctions.fillForward
+import com.databricks.labs.overwatch.pipeline.TransformFunctions._
+import com.databricks.labs.overwatch.utils.SchemaTools.structFromJson
+import com.databricks.labs.overwatch.utils._
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.expressions.{Window, WindowSpec}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{Column, DataFrame}
+
+object DbsqlTransforms extends SparkSessionWrapper {
+
+  import spark.implicits._
+  private val logger: Logger = Logger.getLogger(this.getClass)
+  val responseSuccessFilter: Column = $"response.statusCode" === 200
+
+  /**
+   * BEGIN DBSQL generic functions
+   */
+
+  private val auditBaseCols: Array[Column] = Array(
+    'timestamp, 'date, 'organization_id, 'serviceName, 'actionName,
+    $"userIdentity.email".alias("userEmail"), 'requestId, 'response)
+
+  def deriveWarehouseId(): Column = {
+    when(('actionName === "createEndpoint" || 'actionName === "createWarehouse")
+      && responseSuccessFilter,
+      get_json_object($"response.result", "$.id"))
+      .otherwise('id)
+  }
+
+  def deriveWarehouseBase(auditRawDF: DataFrame): DataFrame = {
+    val warehouse_id_gen_w = Window.partitionBy('organization_id, 'warehouse_name)
+      .orderBy('timestamp).rowsBetween(Window.currentRow, 1000)
+    val warehouse_name_gen_w = Window.partitionBy('organization_id, 'warehouse_id)
+      .orderBy('timestamp).rowsBetween(Window.currentRow, 1000)
+
+    val warehouseSummaryCols = auditBaseCols ++ Array[Column](
+      deriveWarehouseId.alias("warehouse_id"),
+      'name.alias("warehouse_name"),
+      'cluster_size,
+      'min_num_clusters.cast("long"),
+      'max_num_clusters.cast("long"),
+      'auto_stop_mins.cast("long"),
+      'spot_instance_policy,
+      'enable_photon.cast("boolean"),
+      'channel,
```

Comment:
suggest `get_json_object('channel, "$.name")` as that seems to be the only populated value at this time.